src/apps/meshavg.C

```diff
@@ -73,7 +72,7 @@ int main(int argc, char ** argv)
                EquationSystems::READ_BASIC_ONLY);
       libMesh::out << "Loaded next solution " << argv[i] << std::endl;
 
-      for (unsigned int s = 0; s != sysnames.size(); ++s)
+      for (auto s : index_range(sysnames))
```

Comment:
This patch is going to be full of -Wconversion warning regressions, isn't it?

I'd say merge it anyway, though.  I never got that cleanup finished and IIRC a proper cleanup would end up including a 99.9%-backwards-compatible-but-damn-it-MOOSE-uses-the-other-0.1%-too API change.

Reply:
> This patch is going to be full of -Wconversion warning regressions, isn't it?

Damn, I really hope not. Let me try compiling with that flag turned on, but what about the new code do you think would trigger the warning?

Reply:
We have a number of places where our APIs take "unsigned int", and allowing C++ to silently downcast a size_t to that can give a warning.  I noticed some more that really stuck out later - anywhere you see me using ```cast_int<unsigned int>``` or ```IntRange<unsigned int>``` in a for loop, you can rest assured I didn't write longer code specifying a smaller int type for fun.  :-P

But like I said, don't worry about it.  This isn't like our paranoid warnings where we got everything ship-shape and plan to keep it that way forever, this is a warning that we've never fully avoided and that I just like to check on every year or so to help myself look for potential 64-bit truncation bugs.